### ISSUE TITLE
twisted.python.test.test_deprecate.TestDeprecationWarnings.test_deprecatedReplacementWithCallable and twisted.python.test.test_deprecate.TestDeprecationWarnings.test_deprecatedReplacement fail under trial -u

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#5197]
|[<img alt="glyph's avatar" src="https://avatars.githubusercontent.com/u/716529?s=50" width="50" height="50">](https://github.com/glyph)| @glyph reported|
|-|-|
|Trac ID|trac#5197|
|Type|defect|
|Created|2011-07-13 06:01:23Z|

```
[FAIL]
Traceback (most recent call last):
  File "twisted/python/test/test_deprecate.py", line 206, in test_deprecatedReplacementWithCallable
    "\n"
twisted.trial.unittest.FailTest: not equal:
a = '\n    Do nothing.\n\n    This is used to test the deprecation decorators.\n\n    Deprecated in Twisted 8.0.0.\n\n    Deprecated in Twisted 8.0.0; please use twisted.python.test.test_deprecate.dummyReplacementMethod instead.\n    '
b = '\n    Do nothing.\n\n    This is used to test the deprecation decorators.\n\n    Deprecated in Twisted 8.0.0; please use twisted.python.test.test_deprecate.dummyReplacementMethod instead.\n    '


twisted.python.test.test_deprecate.TestDeprecationWarnings.test_deprecatedReplacementWithCallable
===============================================================================
[FAIL]
Traceback (most recent call last):
  File "twisted/python/test/test_deprecate.py", line 186, in test_deprecatedReplacement
    "\n"
twisted.trial.unittest.FailTest: not equal:
a = '\n    Do nothing.\n\n    This is used to test the deprecation decorators.\n\n    Deprecated in Twisted 8.0.0.\n\n    Deprecated in Twisted 8.0.0; please use something.foobar instead.\n    '
b = '\n    Do nothing.\n\n    This is used to test the deprecation decorators.\n\n    Deprecated in Twisted 8.0.0; please use something.foobar instead.\n    '


twisted.python.test.test_deprecate.TestDeprecationWarnings.test_deprecatedReplacement
```

Attachments:

* [ticket5197.patch](https://raw.githubusercontent.com/twisted/twistedmatrix.com-trac-attachments/trunk/ticket/498/498aa1acf2df8d9e6ad04db44e2a3cd6f2e78f76/dadc74ba6c579cbdd30960dda547ed218ae89a00.patch) (932 bytes) - added by canibanoglu on 2013-04-30 08:18:26Z - test_deprecatedUpdatesDocstring now has a locally defined function for testing purposes and running trial -u no longer spits out errors.

<details><summary>Searchable metadata</summary>

```
trac-id__5197 5197
type__defect defect
reporter__glyph glyph
priority__normal normal
milestone__ 
branch__ 
branch_author__ 
status__closed closed
resolution__fixed fixed
component__core core
keywords__easy easy
time__1310536883000000 1310536883000000
changetime__1370036898000000 1370036898000000
version__None None
owner__therve therve

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="glyph's avatar" src="https://avatars.githubusercontent.com/u/716529?s=50" width="50" height="50">](https://github.com/glyph)<a name="note_2"></a>|@glyph commented|
|-|-|

These tests only fail when the rest of the test class is run along with them, not either in isolation or together.

**Comment by twisted-trac** [comment]
|[<img alt="glyph's avatar" src="https://avatars.githubusercontent.com/u/716529?s=50" width="50" height="50">](https://github.com/glyph)<a name="note_3"></a>|@glyph commented|
|-|-|

This only happens when they are run after `test_deprecatedUpdatesDocstring`, so that test must be the buggy one.  You can verify with:

```
trial -u twisted.python.test.test_deprecate.TestDeprecationWarnings.test_deprecatedReplacement twisted.python.test.test_deprecate.TestDeprecationWarnings.test_deprecatedUpdatesDocstring
```

**Comment by twisted-trac** [comment]
|[<img alt="Julian's avatar" src="https://avatars.githubusercontent.com/u/329822?s=50" width="50" height="50">](https://github.com/Julian)<a name="note_4"></a>|@Julian commented|
|-|-|

[#6150](https://github.com/twisted/twisted/issues/6150) was closed as a duplicate of this. Also, `--order toptobottom` from [#5787](https://github.com/twisted/twisted/issues/5787) is an easier way to reproduce this.

**Comment by twisted-trac** [comment]
|<img alt="canibanoglu's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|canibanoglu set owner to canibanoglu|
|-|-|

**Comment by twisted-trac** [comment]
|<img alt="canibanoglu's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|canibanoglu removed owner|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="therve's avatar" src="https://avatars.githubusercontent.com/u/159137?s=50" width="50" height="50">](https://github.com/therve)<a name="note_8"></a>|@therve set owner to @therve|
|-|-|

Looks good, applying.

**Comment by twisted-trac** [comment]
|[<img alt="therve's avatar" src="https://avatars.githubusercontent.com/u/159137?s=50" width="50" height="50">](https://github.com/therve)<a name="note_9"></a>|@therve set status to `closed`|
|-|-|

(In [38572]) TestDeprecationWarnings.test_deprecatedUpdatesDocstring now uses a locally
defined function for testing, fixing a conflict with other tests in the module
which was breaking trial --until-failure.

Author: canibanoglu
Reviewer: therve
Fixes: [#5197](https://github.com/twisted/twisted/issues/5197)
